File: Domain/Budget.cs
using Domain;

public class Budget
{
    public Guid Id { get; set; }
    public string Name { get; set; }
    public decimal TotalGrossIncome { get; set; }
    public decimal TotalNetIncome { get; set; }
    public decimal TotalExpense { get; set; }
    public decimal NetAmount { get; set; }
    public List<Income> Incomes { get; set; }
    public List<Expense> Expenses { get; set; }
}

File: client-app/src/app/models/budget.ts
export interface Budget {
    id: string;
    name: string;
    totalNetIncome: number;
    totalExpense: number;
    totalGrossIncome: number;
    netAmount: number;
    incomes: Income[];
    expenses: Expense[];
}

export interface Income {
    id: string;
    source: string;
    grossAmount: number;
    netAmount: number;
    taxPercentage: number;
}

export interface Expense {
    id: string;
    category: string;
    subcategory: string;
    description: string;
    amount: number;
}